OptimizelySDK/Utils/EventTagUtils.cs

```diff
@@ -16,5 +18,65 @@ public static object GetRevenueValue(Dictionary<string, object> eventTags)
 
            return eventTags[REVENUE_EVENT_METRIC_NAME];
         }
+
+        public static object GetNumericValue(Dictionary<string, object> eventTags, ILogger logger  = null)
+        {
+            string debugMessage = string.Empty;
+            bool isCasted = false;
+
+            float refVar = 0;
+
+            if (eventTags == null)
+                debugMessage = "Event tags is undefined.";
+
+            else if (!eventTags.ContainsKey(VALUE_EVENT_METRIC_NAME))
+                debugMessage = "The numeric metric key is not in event tags.";
+
+            else if (eventTags[VALUE_EVENT_METRIC_NAME] == null)
+                debugMessage = "The numeric metric key value is not defined in event tags.";
+
+            else if (eventTags[VALUE_EVENT_METRIC_NAME] is bool)
+                debugMessage = "Provided numeric value is boolean which is an invalid format.";
+
+            else if (!(eventTags[VALUE_EVENT_METRIC_NAME] is int) && !(eventTags[VALUE_EVENT_METRIC_NAME] is string) && !(eventTags[VALUE_EVENT_METRIC_NAME] is float)
+                && !(eventTags[VALUE_EVENT_METRIC_NAME] is decimal)
+                && !(eventTags[VALUE_EVENT_METRIC_NAME] is double)
+                && !(eventTags[VALUE_EVENT_METRIC_NAME] is float))
+                debugMessage = "Numeric metric value is not in integer, float, or string form.";
+
+            else
+            {
+                if (!float.TryParse(eventTags[VALUE_EVENT_METRIC_NAME].ToString(), out refVar))
+                    debugMessage = string.Format("Provided numeric value {0} is in an invalid format.", eventTags[VALUE_EVENT_METRIC_NAME]);
+                else
+                {
+                    if (float.IsInfinity(refVar))
+                        debugMessage = string.Format("Provided numeric value {0} is in an invalid format.", eventTags[VALUE_EVENT_METRIC_NAME]);
+                    else
+                        isCasted = true;
+                }
+            }
+                
```

Comment:
When do you log the debug message?

Reply:
@msohailhussain -- please address this.

Reply:
I checked-in the code, latest code log the message inside if(logger !=null).

Reply:
Yea, but that doesn't log the debug message. Which branch did you check this into?